### PR TITLE
feat: support nested templates

### PR DESCRIPTION
**Pull request by LiusCraft**
[Task: Support nest html template with ParseFiles](https://github.com/goplus/community/issues/63)

default: Template loaded on first access to HTML(pattern: *_yap.html)

### CODE REVIEW

File: classfile.go [hunk 1]
```diff
@@ -82,6 +82,14 @@ func (p *App) Static__2(pattern string, fs http.FileSystem, allowRedirect ...boo
 	p.StaticHttp(pattern, fs, allowRedirect...)
 }
 
+// load go templates
+func (p *App) LoadTemplate(pattern ...string) {
```
[the remark this answers is not in the document]
Comment (LiusCraft) [reply classfile.go:86]: Users do not need to manually load, but they can choose to use this function to change pattern

[the remark this answers is not in the document]
Comment (LiusCraft) [reply classfile.go:86]: ## 1.Scene:
complex structured projects, such as:
1. /home_yap.html
2. /user/info_yap.html
3. /user/edit_yap.html
4. /article/list_yap.html
5. /article/view_yap.html
6. /layout/consumer_yap.html
7. /layout/admin_yap.html

The template structure of the user is relatively complex, so the default loading scheme is not what the user wants. Therefore, the user can set the pattern by using this function
`LoadTempalte("*_yap. html", "/**/*_yap. html")`
or loading all `.templ`(custom file)
`LoadTemplate("*.templ", "/**/*.templ")`

## 2.Other framework approaches
This feature is supported by mains/tream frameworks such as go template and gin themselves
So it is indispensable
go template: https://cs.opensource.google/go/go/+/refs/tags/go1.22.0:src/text/template/helper.go;l=138
gin: https://github.com/gin-gonic/gin/blob/v1.9.1/gin.go#L252

[the remark this answers is not in the document]
Comment (LiusCraft) [reply classfile.go:86]: @xushiwei PTAL

**Comment by LiusCraft** [comment]
@xushiwei PTAL

File: demo/blog/yap/article_yap.html [hunk 2]
```diff
@@ -1,4 +1,10 @@
 <html>
-<head><meta charset="utf-8"/></head>
-<body>Article {{.id}}</body>
-</html>
\ No newline at end of file
+<head>
```
Comment (xushiwei) [inline demo/blog/yap/article_yap.html:2]: Don't change existed demo. Add new demo instead.

File: template.go [hunk 3]
```diff
@@ -29,11 +33,16 @@ import (
 // as unexported by all other clients.
 type Template struct {
 	*template.Template
+	// yap fs directory
+	fs fs.FS
```
Comment (xushiwei) [inline template.go:37]: Why template should have a `fs`?

Comment (LiusCraft) [reply template.go]: deleted

File: demo/blog/yap/article_yap.html (hunk 2 above)
Comment (LiusCraft) [reply demo/blog/yap/article_yap.html]: added

**Pull request merged**